<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Bioinformatics Pipeline for ColocBoost</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Bioinformatics Pipeline for ColocBoost</h1>



<p>This vignette demonstrates how to use the bioinformatics pipeline for
ColocBoost to perform colocalization analysis with
<code>colocboost</code>.</p>
<ul>
<li>See more details about functions in the package <code>pecotmr</code>
with <a href="https://github.com/StatFunGen/pecotmr/tree/main">link</a>
and <code>colocboost_pipeline</code> with <a href="https://github.com/StatFunGen/pecotmr/blob/main/R/colocboost_pipeline.R">link</a>.</li>
<li>See more details about input data preparation in
<code>xqtl_protocol</code> with <a href="https://statfungen.github.io/xqtl-protocol/code/mnm_analysis/mnm_methods/colocboost.html">link</a>.</li>
</ul>
<div id="loading-data-using-colocboost_analysis_pipeline-function" class="section level1">
<h1>1. Loading Data using <code>colocboost_analysis_pipeline</code>
function</h1>
<p>This function harmonizes the input data and prepares it for
colocalization analysis. In this section, we introduce how to load the
regional data required for the ColocBoost analysis using the
<code>load_multitask_regional_data</code> function. This function loads
mixed datasets for a specific region, including individual-level data
(genotype, phenotype, covariate data) or summary statistics (sumstats,
LD). Run <code>load_regional_univariate_data</code> and
<code>load_rss_data</code> multiple times for different datasets.</p>
<p>Below are the input parameters for this function:</p>
<div id="loading-individual-level-data-from-multiple-cohorts" class="section level2">
<h2>1.1. Loading individual-level data from multiple cohorts</h2>
<ul>
<li><strong><code>region</code></strong> (required): A string of
<code>chr:start-end</code> for the phenotype region you want to
analyze.</li>
<li><strong><code>genotype_list</code></strong>: A vector of paths for
PLINK bed files containing genotype data.</li>
<li><strong><code>phenotype_list</code></strong>: A vector of paths for
phenotype file names.</li>
<li><strong><code>covariate_list</code></strong>: A vector of paths for
covariate file names corresponding to the phenotype file vector.</li>
<li><strong><code>conditions_list_individual</code></strong>: A vector
of strings representing different conditions or groups.</li>
<li><strong><code>match_geno_pheno</code></strong>: A vector of indices
of phenotypes matched to genotype if multiple genotype PLINK files are
used.</li>
<li><strong><code>maf_cutoff</code></strong>: Minimum minor allele
frequency (MAF) cutoff. Default is 0.</li>
<li><strong><code>mac_cutoff</code></strong>: Minimum minor allele count
(MAC) cutoff. Default is 0.</li>
<li><strong><code>xvar_cutoff</code></strong>: Minimum variance cutoff.
Default is 0.</li>
<li><strong><code>imiss_cutoff</code></strong>: Maximum individual
missingness cutoff. Default is 0.</li>
<li><strong><code>association_window</code></strong>: A string of
<code>chr:start-end</code> for the association analysis window (cis or
trans). If not provided, all genotype data will be loaded.</li>
<li><strong><code>extract_region_name</code></strong>: A list of vectors
of strings (e.g., gene ID <code>ENSG00000269699</code>) to subset the
information when there are multiple regions available. Default is
<code>NULL</code>.</li>
<li><strong><code>region_name_col</code></strong>: Column name
containing the region name. Default is <code>NULL</code>.</li>
<li><strong><code>keep_indel</code></strong>: Logical indicating whether
to keep insertions/deletions (INDELs). Default is
<code>TRUE</code>.</li>
<li><strong><code>keep_samples</code></strong>: A vector of sample names
to keep. Default is <code>NULL</code>.</li>
</ul>
<p><strong>Illustrated example</strong></p>
<p>The following example demonstrates how to set up input data with 3
phenotypes and 2 cohorts, where the first cohort has 2 phenotypes and
the second cohort has 1 phenotype.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Example of loading individual-level data</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>region <span class="ot">=</span> <span class="st">&quot;chr1:1000000-2000000&quot;</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>genotype_list <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;plink_cohort1.1.bed&quot;</span>, <span class="st">&quot;plink_cohort1.2.bed&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>phenotype_list <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;phenotype1_cohort1.bed.gz&quot;</span>, <span class="st">&quot;phenotype2_cohort1.bed.gz&quot;</span>, <span class="st">&quot;phenotype1_cohort2.bed.gz&quot;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>covariate_list <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;covariate1_cohort1.bed.gz&quot;</span>, <span class="st">&quot;covariate2_cohort1.bed.gz&quot;</span>, <span class="st">&quot;covariate1_cohort2.bed.gz&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>conditions_list_individual <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;phenotype1_cohort1&quot;</span>, <span class="st">&quot;phenotype2_cohort1&quot;</span>, <span class="st">&quot;phenotype1_cohort2&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>match_geno_pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>) <span class="co"># indices of phenotypes matched to genotype</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>association_window <span class="ot">=</span> <span class="st">&quot;chr1:1000000-2000000&quot;</span> <span class="co"># same as region for cis-analysis</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># Following parameters need to be set according to your data</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>maf_cutoff <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>mac_cutoff <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>xvar_cutoff <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>imiss_cutoff <span class="ot">=</span> <span class="fl">0.9</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="co"># More advanced parameters see pecotmr::load_multitask_regional_data()</span></span></code></pre></div>
</div>
<div id="loading-summary-statistics-from-multiple-cohorts-or-datasets" class="section level2">
<h2>1.2. Loading summary statistics from multiple cohorts or
datasets</h2>
<ul>
<li><strong><code>sumstat_path_list</code></strong>: A vector of file
paths to the summary statistics.</li>
<li><strong><code>column_file_path_list</code></strong>: A vector of
file paths to the column mapping files.</li>
<li><strong><code>LD_meta_file_path_list</code></strong>: A vector of
paths to LD metadata files. LD metadata is a data frame specifying LD
blocks with columns “chrom”, “start”, “end”, and “path”. “start” and
“end” denote the positions of LD blocks, and “path” is the path of each
LD block, optionally including <code>bim</code> file paths.</li>
<li><strong><code>match_LD_sumstat</code></strong>: Logical indicating
whether to match LD blocks with summary statistics.</li>
<li><strong><code>conditions_list_sumstat</code></strong>: A vector of
strings representing different sumstats.</li>
<li><strong><code>n_samples</code></strong>: User-specified sample size.
If unknown, set as 0 to retrieve from the sumstat file.</li>
<li><strong><code>n_cases</code></strong>: User-specified number of
cases.</li>
<li><strong><code>n_controls</code></strong>: User-specified number of
controls.</li>
<li><strong><code>region</code></strong>: The region where tabix is used
to subset the input dataset.</li>
<li><strong><code>extract_sumstats_region_name</code></strong>:
User-specified gene/phenotype name used to further subset the phenotype
data.</li>
<li><strong><code>sumstats_region_name_col</code></strong>: Filter this
specific column for the <code>extract_sumstats_region_name</code>.</li>
<li><strong><code>comment_string</code></strong>: Comment sign in the
column mapping file, default is <code>#</code>.</li>
<li><strong><code>extract_coordinates</code></strong>: Optional data
frame with columns “chrom” and “pos” for specific coordinates
extraction.</li>
</ul>
<p><strong>Illustrated example</strong></p>
<p>The following example demonstrates how to set up input data with 2
summary statistics and one LD reference.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Example of loading summary statistics</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>sumstat_path_list <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;sumstat1.tsv.gz&quot;</span>, <span class="st">&quot;sumstat2.tsv.gz&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>column_file_path_list <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;mapping_columns_1.yml&quot;</span>, <span class="st">&quot;mapping_columns_2.yml&quot;</span>)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>LD_meta_file_path_list <span class="ot">=</span> <span class="st">&quot;ld_meta_file.tsv&quot;</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>covariate_list <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;covariate1_cohort1.bed.gz&quot;</span>, <span class="st">&quot;covariate2_cohort1.bed.gz&quot;</span>, <span class="st">&quot;covariate1_cohort2.bed.gz&quot;</span>)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>conditions_list_sumstat <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;sumstat_1&quot;</span>, <span class="st">&quot;sumstat_2&quot;</span>)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co"># Following parameters need to be set according to your data</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>n_samples <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>n_cases <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">10000</span>, <span class="dv">20000</span>)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>n_controls <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">20000</span>, <span class="dv">40000</span>)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co"># More advanced parameters see pecotmr::load_multitask_regional_data()</span></span></code></pre></div>
</div>
</div>
<div id="perform-colocboost-using-colocboost_analysis_pipeline-function" class="section level1">
<h1>2. Perform ColocBoost using
<code>colocboost_analysis_pipeline</code> function</h1>
<p>In this section, we perform the colocalization analysis using the
<code>colocboost_analysis_pipeline</code> function. Below are the input
parameters for this function:</p>
<ul>
<li><strong><code>region_data</code></strong>: The output of the
<code>load_multitask_regional_data</code> function.</li>
<li><strong><code>focal_trait</code></strong>: Name of the trait if
performing disease-prioritized ColocBoost.</li>
<li><strong><code>event_filters</code></strong>: A list of patterns for
filtering events based on context names. Example: for sQTL,
<code>list(type_pattern = &quot;.*clu_(\\d+_[+-?]).*&quot;, valid_pattern = &quot;clu_(\\d+_[+-?]):PR:&quot;, exclude_pattern = &quot;clu_(\\d+_[+-?]):IN:&quot;)</code>.</li>
<li><strong><code>maf_cutoff</code></strong>: A scalar to remove
variants with maf &lt; maf_cutoff, default is 0.005.</li>
<li><strong><code>pip_cutoff_to_skip_ind</code></strong>: A vector of
cutoff values for skipping analysis based on PIP values for each
context. Default is 0.</li>
<li><strong><code>pip_cutoff_to_skip_sumstat</code></strong>: A vector
of cutoff values for skipping analysis based on PIP values for each
sumstat. Default is 0.</li>
<li><strong><code>qc_method</code></strong>: Quality control method to
use. Options are “rss_qc”, “dentist”, or “slalom” (default:
“rss_qc”).</li>
<li><strong><code>impute</code></strong>: Logical; if TRUE, performs
imputation for outliers identified in the analysis (default: TRUE).</li>
<li><strong><code>impute_opts</code></strong>: A list of imputation
options including rcond, R2_threshold, and minimum_ld (default:
<code>list(rcond = 0.01, R2_threshold = 0.6, minimum_ld = 5)</code>).</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># region_data &lt;- load_multitask_regional_data(...)</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co"># res &lt;- colocboost_analysis_pipeline(region_data)</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
