<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Input Data Format</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Input Data Format</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(colocboost)</span></code></pre></div>
<p>This vignette documents the standard input data formats of
<code>colocboost</code>.</p>
<div id="individual-level-data" class="section level1">
<h1>1. Individual Level Data</h1>
<p>For analyses using individual-level data, the basic format for single
trait is as follows:</p>
<ul>
<li><code>X</code> is an <span class="math inline">\(N \times P\)</span>
matrix with <span class="math inline">\(N\)</span> individuals and <span class="math inline">\(P\)</span> variants. Including variant names as
column names is highly recommended, especially when working with
multiple <span class="math inline">\(X\)</span> matrices and <span class="math inline">\(Y\)</span> vectors.</li>
<li><code>Y</code> is a length <span class="math inline">\(N\)</span>
vector containing phenotype values for the same <span class="math inline">\(N\)</span> individuals as <span class="math inline">\(X\)</span>.</li>
</ul>
<p>The input format for multiple traits is similar, but <code>X</code>
should be a list of genotype matrices, each corresponding to a different
trait. <code>Y</code> should also be a list of phenotype vectors. For
example:</p>
<ul>
<li><code>X = list(X1, X2, X3, X4, X5)</code> where each <code>Xi</code>
is a matrix for trait <code>i</code> - with the dimension of <span class="math inline">\(N_i \times P_i\)</span>, where <span class="math inline">\(N_i\)</span> and <span class="math inline">\(P_i\)</span> do not need to be the same for
different traits.</li>
<li><code>Y = list(Y1, Y2, Y3, Y4, Y5)</code> where each <code>Yi</code>
is a vector for trait <code>i</code> - with <span class="math inline">\(N_i\)</span> individuals.</li>
</ul>
<p><code>colocboost</code> also offers flexible input options (see
detailed usage with different input formats, refer to <a href="https://statfungen.github.io/colocboost/articles/Individual_Level_Colocalization.html">Individual
Level Data Colocalization</a>):</p>
<ul>
<li>Single <span class="math inline">\(X\)</span> matrix with <span class="math inline">\(N \times P\)</span>, and <span class="math inline">\(Y\)</span> matrix with <span class="math inline">\(N \times L\)</span> for <span class="math inline">\(L\)</span> traits.</li>
<li>Multiple <span class="math inline">\(X\)</span> matrices and
unmatched <span class="math inline">\(Y\)</span> vectors with a mapping
dictionary (example shown in section 3 below).</li>
</ul>
</div>
<div id="summary-statistics" class="section level1">
<h1>2. Summary Statistics</h1>
<p>For analyses using summary statistics, the basic format for single
trait is as follows:</p>
<ul>
<li><code>sumstat</code> is a data frame with required columns
<code>z</code> or (<code>beta</code>, <code>sebeta</code>), and optional
columns but highly recommended <code>n</code> and
<code>variant</code>.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">data</span>(Sumstat_5traits)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">head</span>(Sumstat_5traits<span class="sc">$</span>sumstat[[<span class="dv">1</span>]])</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt;              z    n variant</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; 451 -1.0945531 1153    rs_1</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; 452 -0.4113347 1153    rs_2</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; 453 -0.4113347 1153    rs_3</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; 454 -0.7467923 1153    rs_4</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; 455 -0.3018575 1153    rs_5</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; 456 -0.5256479 1153    rs_6</span></span></code></pre></div>
<pre><code>- `z` or (`beta`, `sebeta`) - required: either z-score or (effect size and standard error)
- `n` - highly recommended: sample size for the summary statistics, it is highly recommendation to provide.
- `variant` - highly recommended: required if sumstat for different outcomes do not have the same number of variables (multiple sumstat and multiple LD).</code></pre>
<ul>
<li><code>LD</code> is a matrix of LD. This matrix does not need to
contain the exact same variants as in <code>sumstat</code>, but the
<code>colnames</code> and <code>rownames</code> of <code>LD</code>
should include the <code>variant</code> names for proper alignment.</li>
</ul>
<p>The input format for multiple traits is similar, but
<code>sumstat</code> should be a list of data frames
<code>sumstat = list(sumstat1, sumstat2, sumstat3)</code>. The
flexibility of input format for multiple traits is as follows (see
detailed usage with different input formats, refer to <a href="https://statfungen.github.io/colocboost/articles/Summary_Level_Colocalization.html">Summary
Statistics Colocalization</a>):</p>
<ul>
<li>One LD matrix with a superset of variants in <code>sumstat</code>
for all traits is allowed.</li>
<li>Multiple LD matrices, each corresponding to a different trait, are
also allowed for the trait-specific LD structure.</li>
<li>Multiple LD matrices and unmatched <code>sumstat</code> data frames
with a mapping dictionary are also allowed (example shown in section 3
below).</li>
</ul>
</div>
<div id="optional-mapping-between-arbitrary-input-x-and-y" class="section level1">
<h1>3. Optional: mapping between arbitrary input <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span></h1>
<p>For analysis when including multiple genotype matrices <code>X</code>
with unmatched arbitrary phenotype vectors <code>Y</code>, a mapping
dictionary <code>dict_YX</code> is required to indicate the relationship
between <code>X</code> and <code>Y</code>. Similarly, when multiple LD
matrices with unmatched arbitrary multiple summary statistics
<code>sumstat</code> are used, a mapping dictionary
<code>dict_sumstatLD</code> is required to indicate the relationship
between <code>sumstat</code> and <code>LD</code>.</p>
<p>For example, considering three genotype matrices
<code>X = list(X1, X2, X3)</code> and 6 phenotype vectors
<code>Y = list(Y1, Y2, Y3, Y4, Y5, Y6)</code>, where</p>
<ul>
<li><code>X1</code> is for trait 1, trait 2, trait 3</li>
<li><code>X2</code> is for trait 4, trait 5</li>
<li><code>X3</code> is for trait 6</li>
</ul>
<p>Then, you need to define a 6 by 2 matrix mapping dictionary
<code>dict_YX</code> as follows:</p>
<ul>
<li>The first column should be <code>c(1,2,3,4,5,6)</code> for 6
traits.</li>
<li>The second column should be <code>c(1,1,1,2,2,3)</code> for 3
genotype matrices.</li>
</ul>
<p>Here, each row indicates the trait index and the corresponding
genotype matrix index.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>dict_YX <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>dict_YX</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt;      [,1] [,2]</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt; [1,]    1    1</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; [2,]    2    1</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; [3,]    3    1</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; [4,]    4    2</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; [5,]    5    2</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; [6,]    6    3</span></span></code></pre></div>
</div>
<div id="hyprcoloc-compatible-format-effect-size-and-standard-error-matrices" class="section level1">
<h1>4. HyPrColoc compatible format: effect size and standard error
matrices</h1>
<p>ColocBoost also provides a flexibility to use HyPrColoc compatible
format for summary statistics with and without LD matrix. For example,
when analyze <span class="math inline">\(L\)</span> traits for the same
<span class="math inline">\(P\)</span> variants with the specified
effect size and standard error matrices:</p>
<ul>
<li><code>effect_est</code> (required) is <span class="math inline">\(P
\times L\)</span> matrix of variable regression coefficients
(i.e. regression beta values) in the genomic region.</li>
<li><code>effect_se</code> (required) is <span class="math inline">\(P
\times L\)</span> matrix of standard errors for the regression
coefficients.</li>
<li><code>effect_n</code> (highly recommended) is either a scalar or a
vector of sample sizes for estimating regression coefficients.</li>
<li><code>LD</code> (optional) is LD matrix for the <span class="math inline">\(P\)</span> variants. If it is not provided, it
will apply LD-free ColocBoost.</li>
</ul>
<p>See more details about HyPrColoc compatible format in <a href="https://statfungen.github.io/colocboost/articles/Summary_Level_Colocalization.html">Summary
Statistics Colocalization</a>).</p>
<p>See more details about data format to implement LD-free ColocBoost
and LD-mismatch diagnosis in <a href="https://statfungen.github.io/colocboost/articles/LD_Free_Colocalization.html">LD
mismatch and LD-free Colocalization</a>).</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
