<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>LD mismatch and LD-free Colocalization</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">LD mismatch and LD-free Colocalization</h1>



<p>This vignette demonstrates LD mismatch diagnosis in the
<code>colocboost</code> package and how to perform LD-mismatch and
LD-free colocalization analysis, when some traits completely lack LD
information or share only partial variant coverage with other
traits.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(colocboost)</span></code></pre></div>
<div id="ld-mismatch-diagnosis" class="section level1">
<h1>1. LD mismatch diagnosis</h1>
<p>The <code>colocboost</code> assumes that the LD matrix accurately
estimates the correlations among variants from the original GWAS
genotype data. Typically, the LD matrix comes from some public databases
of genotypes in a suitable reference population. An inaccurate LD matrix
may lead to unreliable colocalization results, especially if the LD
matrix is significantly different from the one estimated from the
original genotype data.</p>
<div id="why-ld-mismatch-matters" class="section level3">
<h3>Why LD Mismatch Matters</h3>
<p>An inaccurate LD matrix can cause inconsistencies between the summary
statistics and the reference LD matrix, leading to:</p>
<ul>
<li>Biased estimates of causal variants.</li>
<li>Increased computational time due to slower algorithm
convergence.</li>
<li>Potentially misleading colocalization results.</li>
</ul>
<p>ColocBoost provides diagnostic warnings for assessing the consistency
of the summary statistics with the reference LD matrix.</p>
<ul>
<li>Estimated residual variance of the model is negative or greater than
phenotypic variance (<code>rtr &lt; 0</code> or
<code>rtr &gt; var_y</code>; see details in Supplementary Note
S3.5.2).</li>
<li>The trait-specific gradient boosting model fails to converge.</li>
<li>Change in trait-specific profile log-likelihood according to a CoS
is negative (see details in Supplementary Note S3.5.3).</li>
</ul>
</div>
<div id="example-of-including-ld-mismatch" class="section level3">
<h3>Example of including LD mismatch</h3>
<p>In this example, we create a simulated dataset with LD mismatch by
changing the sign of Z-scores for 1% of variants for each trait.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Create a simulated dataset with LD mismatch</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;Sumstat_5traits&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;Ind_5traits&quot;</span>)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>LD <span class="ot">&lt;-</span> <span class="fu">get_cormat</span>(Ind_5traits<span class="sc">$</span>X[[<span class="dv">1</span>]])</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co"># Change sign of Z-score for 1% of variants for each trait by including mismatched LD</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>miss_prop <span class="ot">&lt;-</span> <span class="fl">0.005</span> </span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>sumstat <span class="ot">&lt;-</span> <span class="fu">lapply</span>(Sumstat_5traits<span class="sc">$</span>sumstat, <span class="cf">function</span>(ss){</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">nrow</span>(ss)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>  pos_miss <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span>p, <span class="fu">ceiling</span>(miss_prop <span class="sc">*</span> p))</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>  ss<span class="sc">$</span>z[pos_miss] <span class="ot">&lt;-</span> <span class="sc">-</span>ss<span class="sc">$</span>z[pos_miss]</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>  <span class="fu">return</span>(ss)</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>})</span></code></pre></div>
</div>
<div id="running-colocboost-with-ld-mismatch" class="section level3">
<h3>Running ColocBoost with LD Mismatch</h3>
<p>When running <code>colocboost</code> with an LD mismatch, you may
encounter diagnostic warnings. These warnings are not errors, and the
analysis will still proceed. However, the results may be less reliable
due to the mismatch, and the computational time may increase as the
algorithm takes longer to converge.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">colocboost</span>(<span class="at">sumstat =</span> sumstat, <span class="at">LD =</span> LD)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">#&gt; Validating input data.</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt; Starting gradient boosting algorithm.</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; Warning in colocboost_workhorse(cb_data, M = M, prioritize_jkstar =</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; prioritize_jkstar, : ColocBoost gradient boosting for outcome 5 did not</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; coverage in 500 iterations! Please check consistency between summary statistics</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt; and LD matrix. See details in tutorial website</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt; https://statfungen.github.io/colocboost/articles/.</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; Gradient boosting for outcome 4 converged after 523 iterations!</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; Gradient boosting at 1000 iterations, still updating.</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; Warning in colocboost_workhorse(cb_data, M = M, prioritize_jkstar =</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; prioritize_jkstar, : ColocBoost gradient boosting for outcome 1 did not</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; coverage in 500 iterations! Please check consistency between summary statistics</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; and LD matrix. See details in tutorial website</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co">#&gt; https://statfungen.github.io/colocboost/articles/.</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co">#&gt; Gradient boosting for outcome 2  stop since rtr &lt; 0 or max(correlation) &gt; 1 after 1213 iterations! Results for this locus are not stable, please check if mismatch between sumstat and LD! See details in tutorial website https://statfungen.github.io/colocboost/articles/.</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#&gt; Gradient boosting for outcome 3  stop since rtr &lt; 0 or max(correlation) &gt; 1 after 1475 iterations! Results for this locus are not stable, please check if mismatch between sumstat and LD! See details in tutorial website https://statfungen.github.io/colocboost/articles/.</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co">#&gt; Performing inference on colocalization events.</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt; Warning in get_cos_profile(cs_beta, outcome_idx, X = cb_data$data[[X_dict]]$X,</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#&gt; : Warning message: potential sumstat &amp; LD mismatch may happens for outcome 2 .</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co">#&gt; Using logLR = CoS(profile) - max(profile). Please check our website</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#&gt; https://statfungen.github.io/colocboost/articles/.</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#&gt; Extracting colocalization results with pvalue_cutoff = 0.001, cos_npc_cutoff = 0.2, and npc_outcome_cutoff = 0.2.</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt; Keep only CoS with cos_npc &gt;= 0.2. For each CoS, keep the outcomes configurations that pvalue of variants for the outcome &lt; 0.001 and npc_outcome &gt;0.2.</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a>res<span class="sc">$</span>cos_details<span class="sc">$</span>cos<span class="sc">$</span>cos_index</span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt; $`cos1:y1_y3_y4`</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co">#&gt; [1] 186 205 194 168 218 228 229</span></span></code></pre></div>
<p>These warnings serve as diagnostic tools to alert users about
potential inconsistencies in the input data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>res<span class="sc">$</span>cos_details<span class="sc">$</span>cos_outcomes_npc</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#&gt; $`cos1:y1_y2_y3_y4`</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt;    outcomes_index relative_logLR npc_outcome</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt; Y3              3      2.3111393   0.9901696</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; Y1              1      1.9710178   0.9805913</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; Y4              4      0.9895934   0.8618184</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; Y2              2      0.0000000   0.0000000</span></span></code></pre></div>
<p><strong>Note</strong>: In the above example, the normalized
probability of trait 2 is 0, indicating that colocalization with trait 2
may be less reliable due to the LD mismatch. This is a warning, not an
error, and the colocalization analysis will still proceed. Therefore, in
this case, we suggest treating the colocalization of trait 2 with
caution.</p>
<p>Potential solutions include:</p>
<ul>
<li>Using a more accurate LD matrix for trait 2.</li>
<li>Excluding trait 2 from the analysis or colocalization results.</li>
<li>Using the LD-mismatch or LD-free approach for colocalization
analysis (below).</li>
<li>Check the visualization of colocalization results including trait 2
using <code>colocboost_plot(res)</code>. Remove the potential spurious
signals when LD mismatch is detected using
<ul>
<li><code>get_robust_colocalization(res, cos_npc_cutoff = 0.5, npc_outcome_cutoff = 0.2)</code>
to exclude the trait 2 in the above example if the signals are not
reasonable.</li>
<li><code>get_robust_colocalization(res, pvalue_cutoff = 1e-5, cos_npc_cutoff = 0, npc_outcome_cutoff = 0)</code>
to include all colocalized traits with the larger marginal evidence, but
the mismatch is detected.</li>
</ul></li>
</ul>
</div>
</div>
<div id="ld-free-and-ld-mismatch-colocalization-analysis" class="section level1">
<h1>2. LD-free and LD-mismatch colocalization analysis</h1>
<p>When there is substantial discordance between the LD matrix and
summary statistics, the reliability of colocalization analysis may be
compromised. Such discordance can arise when the LD matrix and summary
statistics are derived from different populations or when the LD matrix
is estimated from a smaller or less representative reference sample.
This can lead to unexpected results, such as biased causal variant
identification or reduced accuracy in the analysis.</p>
<p>To address these challenges, ColocBoost provides two alternative
approaches for colocalization analysis with the assumption of one causal
variant per trait per region:</p>
<ul>
<li><p><strong>One iteration approach</strong> (recommended): performing
only 1 iteration of gradient boosting with the LD matrix ensures
that:</p>
<ul>
<li>The LD matrix is only used to check the equivalence among
trait-specific best update variants.</li>
<li>The accuracy of the results is improved compared to completely
ignoring the LD matrix.</li>
</ul></li>
</ul>
<p>This method is particularly useful when the LD matrix is mismatched
but still provides valuable insights into variant correlations.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># Perform only 1 iteration of gradient boosting with LD matrix</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>res_mismatch <span class="ot">&lt;-</span> <span class="fu">colocboost</span>(<span class="at">sumstat =</span> sumstat, <span class="at">LD =</span> LD, <span class="at">M =</span> <span class="dv">1</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#&gt; Validating input data.</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt; Starting gradient boosting algorithm.</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt; Running ColocBoost with assumption of one causal per outcome per region!</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; Performing inference on colocalization events.</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; Extracting colocalization results with pvalue_cutoff = 0.001, cos_npc_cutoff = 0.2, and npc_outcome_cutoff = 0.2.</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; Keep only CoS with cos_npc &gt;= 0.2. For each CoS, keep the outcomes configurations that pvalue of variants for the outcome &lt; 0.001 and npc_outcome &gt;0.2.</span></span></code></pre></div>
<ul>
<li><strong>LD-free</strong>: when the mismatch between the LD matrix
and summary statistics is too large to be useful or when LD information
is completely unavailable, ColocBoost provides an LD-free approach.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>res_free <span class="ot">&lt;-</span> <span class="fu">colocboost</span>(<span class="at">sumstat =</span> sumstat)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#&gt; Validating input data.</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; Warning in colocboost_validate_input_data(X = X, Y = Y, sumstat = sumstat, :</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; Providing the LD for summary statistics data is highly recommended. Without LD,</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; only a single iteration will be performed under the assumption of one causal</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; variable per outcome. Additionally, the purity of CoS cannot be evaluated!</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; Starting gradient boosting algorithm.</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; Running ColocBoost with assumption of one causal per outcome per region!</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; Performing inference on colocalization events.</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; Extracting colocalization results with pvalue_cutoff = 0.001, cos_npc_cutoff = 0.2, and npc_outcome_cutoff = 0.2.</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; Keep only CoS with cos_npc &gt;= 0.2. For each CoS, keep the outcomes configurations that pvalue of variants for the outcome &lt; 0.001 and npc_outcome &gt;0.2.</span></span></code></pre></div>
<p>While this method is computationally efficient, it has limitations
due to the strong assumption of a single causal variant per trait per
region. Users should interpret the results with caution, especially in
regions with complex LD structures or multiple causal variants.</p>
<p>ColocBoost also provides a flexibility to use HyPrColoc compatible
format for summary statistics without LD matrix.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># Loading the Dataset</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">data</span>(Ind_5traits)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>X <span class="ot">&lt;-</span> Ind_5traits<span class="sc">$</span>X</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>Y <span class="ot">&lt;-</span> Ind_5traits<span class="sc">$</span>Y</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co"># Coverting to HyPrColoc compatible format</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>effect_est <span class="ot">&lt;-</span> effect_se <span class="ot">&lt;-</span> effect_n <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(X)){</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>  x <span class="ot">&lt;-</span> X[[i]]</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>  y <span class="ot">&lt;-</span> Y[[i]]</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>  effect_n[i] <span class="ot">&lt;-</span> <span class="fu">length</span>(y)</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>  output <span class="ot">&lt;-</span> susieR<span class="sc">::</span><span class="fu">univariate_regression</span>(<span class="at">X =</span> x, <span class="at">y =</span> y)</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>  effect_est <span class="ot">&lt;-</span> <span class="fu">cbind</span>(effect_est, output<span class="sc">$</span>beta)</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>  effect_se <span class="ot">&lt;-</span> <span class="fu">cbind</span>(effect_se, output<span class="sc">$</span>sebeta)</span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>}</span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="fu">colnames</span>(effect_est) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(effect_se) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Y1&quot;</span>, <span class="st">&quot;Y2&quot;</span>, <span class="st">&quot;Y3&quot;</span>, <span class="st">&quot;Y4&quot;</span>, <span class="st">&quot;Y5&quot;</span>)</span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="fu">rownames</span>(effect_est) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(effect_se) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X[[<span class="dv">1</span>]])</span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="co"># Run colocboost</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">colocboost</span>(<span class="at">effect_est =</span> effect_est, <span class="at">effect_se =</span> effect_se, <span class="at">effect_n =</span> effect_n)</span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a><span class="co">#&gt; Validating input data.</span></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a><span class="co">#&gt; Warning in colocboost_validate_input_data(X = X, Y = Y, sumstat = sumstat, :</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a><span class="co">#&gt; Providing the LD for summary statistics data is highly recommended. Without LD,</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a><span class="co">#&gt; only a single iteration will be performed under the assumption of one causal</span></span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a><span class="co">#&gt; variable per outcome. Additionally, the purity of CoS cannot be evaluated!</span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a><span class="co">#&gt; Starting gradient boosting algorithm.</span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a><span class="co">#&gt; Running ColocBoost with assumption of one causal per outcome per region!</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a><span class="co">#&gt; Performing inference on colocalization events.</span></span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a><span class="co">#&gt; Extracting colocalization results with pvalue_cutoff = 0.001, cos_npc_cutoff = 0.2, and npc_outcome_cutoff = 0.2.</span></span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a><span class="co">#&gt; Keep only CoS with cos_npc &gt;= 0.2. For each CoS, keep the outcomes configurations that pvalue of variants for the outcome &lt; 0.001 and npc_outcome &gt;0.2.</span></span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a><span class="co"># Identified CoS</span></span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a>res<span class="sc">$</span>cos_details<span class="sc">$</span>cos<span class="sc">$</span>cos_index</span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a><span class="co">#&gt; $`cos1:y1_y3_y4`</span></span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a><span class="co">#&gt; [1] 186 205 194 168</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
